File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {

    map1 := make(map[int]bool)
    map2 := make(map[int]bool)

    for i := 0; i < len(nums1); i++ {
        map1[nums1[i]] = true
    }

    for i := 0; i < len(nums2); i++ {
        map2[nums2[i]] = true
    }

    result := []int{}

    for k := range map1 {
        if _, present := map2[k]; present == true {
            result = append(result, k)
        }
    }

    return result
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> numSet;
        for (int v : nums1) numSet.insert(v);
        vector<int> res;
        int prev = -1;
        sort(nums2.begin(), nums2.end());
        for (int i = 0; i < nums2.size(); ++i) {
            if (nums2[i] == prev) {
                continue;
            }
            if (numSet.contains(nums2[i])) {
                res.push_back(nums2[i]);
            }
            prev = nums2[i];
        }
        return res;
    }
};